modules/RTC/JitsiLocalTrack.js

```diff
@@ -136,6 +136,8 @@ export default class JitsiLocalTrack extends JitsiTrack {
         // created (until getConstraints() support), however we can associate
         // tracks with real devices obtained from enumerateDevices() call as
         // soon as it's called.
+        // NOTE: this.deviceId corresponds to the device id specified in GUM constraints and this._realDeviceId seems to
+        // correspond to the id of a matching device from the available device list.
```

Comment:
Did you know why there are two ids? Is it because exact constraints aren't used so another device could be obtained?

Reply:
The device id passed in the constructor is the device id passed to GUM. Currently it serves as fallback if we can't obtain the device id. The use case that you mentioned sounds like one possible reason. Maybe there are other...

Reply:
Tangential question: would relying on label over device id have fixed this issue as well then?

Reply:
What is the issue? That we have 2 variables for device id or that the device id was wrong when the default device is changed?

If you are talking about the default device issue, I'd say no because the issue is that the track uses the old label...

if you are talking about the 2 variables for device id, I'm not sure. I guess we can fix it even if we use the device ids but it will need further investigation...